File: app/controllers/notes_controller.rb
class NotesController < ApplicationController
  include Clearance::Controller

  before_action :require_login
  before_action :set_note, only: [:update, :destroy]
  skip_before_action :verify_authenticity_token

  # GET /notes
  def index
    @notes = current_user.notes.all
    render json: @notes.to_json
  end

  # POST /notes
  def create
    @note = current_user.notes.build(note_params)
    if @note.save
      render json: @note
    else
      # render # error
    end
  end

  # PUT /notes/:id
  def update
    if @note.update(note_params)
      render json: @note
    else
      # render # error
    end
  end

  # DELETE /notes/:id
  def destroy
    if @note.destroy
      head :no_content
    else
      # render # error
    end
  end

  private

    def set_note
      @note = current_user.notes.find(params[:id])
    end

    def note_params
      params.require(:note).permit(:category, :details, :text, :title)
    end

  # render json: params.to_json
end

File: app/assets/javascripts/angular/notes.controller.js
(function() {
    'use strict';

    angular
        .module('app')
        .controller('NotesController', NotesController);

        NotesController.$inject = ['NoteService'];

        function NotesController(NoteService) {
          var vm = this;
          vm.index = index;
          vm.create = create;
          vm.update = update;
          vm.destroy = destroy;

          vm.all = vm.index();

          function index() {
            return NoteService.query();
          }

          function create(note) {
            return new NoteService(note).$save();
          }
          // function create(noteData) {
          //   var note = new NoteService(noteData);
          //   return note.$save();
          // }

          function update(note) {
            return NoteService.update({ id: note.id }, note);
          }
          // function update(noteData) {
          //   var note = new NoteService(noteData);
          //   return NoteService.update({ id: note.id }, note);
          // }

          function destroy(note) {
            return NoteService(note).delete({ id: note.id });
          }
          // function destroy(noteData) {
          //   var note = new NoteService(noteData);
          //   return NoteService(note).delete({ id: note.id });
          // }
        }

})();
